gateway/src/resty/oidc/discovery.lua

```diff
@@ -15,6 +15,9 @@ local _M = { }
 
 local mt = { __index = _M }
 
+-- to be overridden in tests
```

Comment:
Why is this needed? Can't we just use the `http_backend` that `.new()` receives?

Reply:
The issue is that when you use this for example in the policy initializer, then the policy should accept this too. Pushing testing client through all levels of the stack seems wrong.

Reply:
If this is only for testing, another option would be to modify `self.backend` in the tests. Not sure if better.

I get your point of having to pass this through multiple objects.

:+1:

Reply:
Actually, this got me thinking. I think we should offer this in http_ng, so we can stub it globally in the whole test suite.